### PR TITLE
chrome_extensions: Add the profile name to the table

### PR DESCRIPTION
**Pull request by alessandrogario**
This PR adds a new column to the `chrome_extensions` table named `profile`, used to show the name of the Chrome profile that owns each extension.

```
osqueryi --verbose 'SELECT profile, name, version FROM chrome_extensions;'
```
```
+---------------+---------------------+--------------+
| profile       | name                | version      |
+---------------+---------------------+--------------+
| trail_of_bits | Slides              | 0.10         |
| trail_of_bits | Docs                | 0.10         |
| trail_of_bits | Google Drive        | 14.1         |
| trail_of_bits | YouTube             | 4.2.8        |
| trail_of_bits | Sheets              | 1.2          |
| trail_of_bits | HTTPS Everywhere    | 2018.8.22    |
| trail_of_bits | Google Docs Offline | 1.7          |
| trail_of_bits | __MSG_APP_NAME__    | 1.0.0.4      |
| trail_of_bits | Gmail               | 8.1          |
| trail_of_bits | Chrome Media Router | 6818.528.0.0 |
| Person 1      | Slides              | 0.10         |
| Person 1      | Docs                | 0.10         |
| Person 1      | Google Drive        | 14.1         |
| Person 1      | YouTube             | 4.2.8        |
| Person 1      | Sheets              | 1.2          |
| Person 1      | Google Docs Offline | 1.7          |
| Person 1      | __MSG_APP_NAME__    | 1.0.0.4      |
| Person 1      | Gmail               | 8.1          |
| Person 1      | Chrome Media Router | 6818.528.0.0 |
+---------------+---------------------+--------------+
```

### CODE REVIEW

File: osquery/tables/applications/browser_utils.cpp [hunk 1]
```diff
@@ -149,20 +185,32 @@ QueryData genChromeBasedExtensions(QueryContext& context,
...
+        if (extensions.empty()) {
+          continue;
+        }
+
+        auto profile_path = fs::path(profile).parent_path().parent_path();
+
+        std::string profile_name;
+        auto status = getChromeProfileName(profile_name, profile_path);
+        if (!status.ok()) {
+          LOG(WARNING) << status.getMessage();
```
Comment (akindyakov) [inline osquery/tables/applications/browser_utils.cpp]: May I ask you to add a prefix for this log line? Smth like: "Getting chrome profile name failed: " ?

File: osquery/tables/applications/browser_utils.cpp [hunk 2]
```diff
@@ -33,10 +33,45 @@ const std::map<std::string, std::string> kExtensionKeys = {
...
+    json_stream << json_data;
+    pt::read_json(json_stream, tree);
+  } catch (const pt::json_parser::json_parser_error&) {
+    return Status(
+        1, "Failed to parse the Preferences file for profile " + path.string());
+  }
+
+  const auto& profile_obj = tree.get_child_optional(kProfilePreferenceKey);
+  if (!profile_obj) {
+    return Status(1, "The following profile is malformed: " + path.string());
```
Comment (akindyakov) [inline osquery/tables/applications/browser_utils.cpp]: Nit: you could use a bit more self descriptive form of creating status with error:
```c++
return Status::failure(...);
```
If you like so :)

**Comment by guliashvili** [comment]
ok to test

**Comment by guliashvili** [comment]
Hi @alessandrogario 
If you could rebase it on experimental, I will merge it soon.

**Comment by alessandrogario** [comment]
Hello @guliashvili !

I've updated the PR to target experimental! Thanks for looking into this patch!

**Pull request closed without merging**